examples/image_viewer.rs

```diff
@@ -262,85 +251,22 @@ fn main() {
-                        &token,
-                    )
-                };
-                // Now, swap the MemPools, so next time we'll draw on the other.
-                // THis is Double-buffering.
-                pools.swap();
-                // Register that one more pool is in use.
-                free_pools.set(free_pools.get() - 1);
-                // Attach this new buffer to our surface.
-                window.surface().attach(Some(&buffer), 0, 0);
-                // We need to tell the server which part of the surface have changed,
-                // it uses it to only reload/redraw relevant parts, for performance.
-                // In our case, everything has likely changed, so we damage everything.
-                //
-                // Two ways are possible for notifying this damage:
-                if window.surface().version() >= 4 {
-                    // If our server is recent enough and supports at least version 4 of the
-                    // wl_surface interface, we can specify the damage in buffer coordinates.
-                    // This is obviously the best and do that if possible.
-                    window
-                        .surface()
-                        .damage_buffer(0, 0, dimensions.0 as i32, dimensions.1 as i32);
-                } else {
-                    // Otherwise, we fallback to compatilibity mode. Here we specify damage
-                    // in surface coordinates, which would have been different if we had drawn
-                    // our buffer at HiDPI resolution. We didn't though, so it is ok.
-                    // Using `damage_buffer` in general is better though.
-                    window
-                        .surface()
-                        .damage(0, 0, dimensions.0 as i32, dimensions.1 as i32);
-                }
-                // We resize our frame, so that is is draw at the right size by SCTK
-                window.resize(dimensions.0, dimensions.1);
-                // We also refresh it, so that SCTK actually draw it with the new size
-                window.refresh();
-                // We commit our drawing surface, so that the server atomically applies
-                // all the changes we previously did.
-                window.surface().commit();
-                // We don't need to redraw or refresh anymore =)
-                need_redraw = false;
-                need_refresh = false;
+                    if resizing { None } else { Some(&image) },
+                ),
+                None => {}
             }
-        } else if need_refresh {
-            // If we don't need to redraw but only to refresh do it
-            window.refresh();
-            // We still need to commit our surface, as the decorations are drawn as
-            // subsurfaces of it.
+            // We commit our drawing surface, so that the server atomically applies
+            // all the changes we previously did.
             window.surface().commit();
```

Comment:
You moved the `commit()` into the `redraw` function, so I think this line can be removed.